RadarSDK/RadarLocationPremissionsManager.m

```diff
@@ -0,0 +1,167 @@
+#import "RadarDelegateHolder.h"
+
+@implementation RadarLocationPermissionsManager{
+    BOOL danglingBackgroundPermissionsRequest;
+    BOOL inBackgroundLocationPopUp;
+}
+
++ (instancetype)sharedInstance {
+    static dispatch_once_t once;
+    static id sharedInstance;
+    dispatch_once(&once, ^{
+        sharedInstance = [self new];
+    });
+    return sharedInstance;
+}
+
+- (instancetype)init {
+    self = [super init];
+    if (self) {
+        self.locationManager = [CLLocationManager new];
+        self.locationManager.delegate = self;
+        RadarLocationPermissionsStatus *status = [RadarLocationPermissionsStatus retrieve];
+        if (status) {
+            self.status = status;
+            // we need to set the in permissions flag to no as we should not start in the popup state
+            self.status.inForegroundPopup = NO;
+        } else{
+            if (@available(iOS 14.0, *)) {
+                self.status = [[RadarLocationPermissionsStatus alloc] initWithStatus:self.locationManager.authorizationStatus
+                                                            backgroundPopupAvailable:YES
+                                                                   inForegroundPopup:NO
+                                                   userRejectedBackgroundPermissions:NO];
+            } else {
+                // just a dummy value, this component will not communicate with the rest of the SDK if the version is below 14.0
+                self.status = [[RadarLocationPermissionsStatus alloc] initWithStatus: kCLAuthorizationStatusAuthorizedAlways
+                                                            backgroundPopupAvailable:NO
+                                                                   inForegroundPopup:NO
+                                                   userRejectedBackgroundPermissions:NO];
+            }
+        }
+       
+        [[NSNotificationCenter defaultCenter] addObserver:self
+                                                 selector:@selector(applicationWillResignActive)
+                                                     name:UIApplicationWillResignActiveNotification
+                                                   object:nil];
+        [[NSNotificationCenter defaultCenter] addObserver:self
+                                                 selector:@selector(applicationDidBecomeActive)
+                                                     name:UIApplicationDidBecomeActiveNotification
+                                                   object:nil];        
+        
+    }
+    return self;
+}
+
+- (void)updateStatus:(RadarLocationPermissionsStatus *)status {
+    self.status = status;
+    [RadarLocationPermissionsStatus store:status];
+    [self sendDelegateUpdate];
+}
+
+- (void)sendDelegateUpdate {
+    if (@available(iOS 14.0, *)) {
+        [[RadarDelegateHolder sharedInstance] didUpdateLocationPermissionsStatus:self.status];
+    }
+}
```

Comment:
I think you can just pull this into the above, no need to have a function `sendDelegateUpdate`